lib/backend/backend.go

```diff
@@ -248,6 +248,19 @@ type Event struct {
 	Item Item
 }
 
+// IterationOrder dictates the order in which iteration should
+// return items.
+type IterationOrder int
```

Comment:
When adding sort order to access requests I went with a `Descending bool` field.  This translates across the GRPC boundary really nicely without custom type conversion, sidesteps the need to handle invalid values, and IMO is more inkeeping with the general teleport philosophy of treating ascending as the default order.  Consider doing something similar here?